javascript/javascript1/week2/script.js

```diff
@@ -3,13 +3,39 @@
 
 // Flight booking fullname function
 
-function getFullname(firstname, surname) {
-    return firstname + " " + surname;
-};
 
-console.log(getFullname("Anastasiia", "Yaitska"));
+// 1. first part
+
+// function getFullname(firstname, surname) {
+ 
+//     return firstname + " " + surname;
+// };
+
+// console.log(getFullname("Anastasiia", "Yaitska"));
+
+// const fullname1 = "Sascha";
+// const fullname2 = "Nata";
+
+// console.log(getFullname(fullname1, fullname2));
 
-const fullname1 = "Sascha";
-const fullname2 = "Nata";
 
-console.log(getFullname(fullname1, fullname2));
\ No newline at end of file
+// 2. second part
+
+const useFormalName = true;
+const gender = "female"
+
+function getFullname(firstname, surname) {
```

Comment:
The `useFormalName` should have been added as a **parameter** to the function. Thus the function would look like 
```javascript
function getFullName(firstname, surname, useFormalName) {
  ...
}
```

The reason why your function is still correct (without errors) is because it is reading the value of `useFormalName` from the outside of the function. 🤓 So ideally the function should check for empty inputs first: if the `firstname` or `surname` is equal to an empty string, which you are already checking for one of the variables but not the other one. However, you check in the correct way, by writing

```javascript
if (variable === "")
```

This can easily be done incorrectly in JavaScript, so good job that you did it well! 🥇 If one of the inputs is empty, return a message like `Try again`. Another condition should be to check if the `useFormalName` is true, and if it is, add the 'Lord' string in front of the name. This way, even if someone does not provide this parameter, (they call the function with just two parameters) the function will just return the name without the 'Lord' by default. 🧠